packages/story-editor/src/components/library/panes/shopping/productImage.js

```diff
@@ -57,4 +76,31 @@ ProductImage.propTypes = {
   product: PropTypes.object.isRequired,
 };
 
-export default ProductImage;
+function DraggableProductImage({ product, isOnPage }) {
+  const src = product?.productImages[0]?.url || '';
+  return (
+    <ProductImageWrap>
+      <ProductImage product={product} />
+      {!isOnPage && src && (
+        <LibraryMoveable
+          type="product"
+          elementProps={{ product }}
+          cloneElement={StyledImageClone}
+          cloneProps={{
+            loading: 'lazy',
+            decoding: 'async',
+            crossOrigin: 'anonymous',
+            src,
+          }}
+        />
+      )}
+    </ProductImageWrap>
```

Comment:
Products should not be draggable if there are already 6 products on the page

Just like the "Add" button is disabled when that happens.

In other words: When I can't add a product by clicking, I should also not be able to add one by dragging.

Reply:
Added the canAddMore check